WRO-8722: Update in-editor linting guide (#51)
enactjs/eslint-config-enact: docs/index.md

If you want to switch to the strict version of the linting rules, modify your `package.json` file and change the following lines:

```json

>     "lint": "enact lint"

Reviewer: Let's not modify this.

Ever since ESLint 6, global installs of ESLint configs are no longer supported.
To work around this new limitation, while still supporting in-editor linting, we've created a new [eslint-config-enact-proxy](https://github.com/enactjs/eslint-config-enact-proxy) package.
The [eslint-config-enact-proxy](https://github.com/enactjs/eslint-config-enact-proxy) acts like a small proxy config, redirecting ESLint to use a globally-installed Enact ESLint config.

> In order for in-editor linting, `eslint-config-enact-proxy` should be installed locally on a project:

Reviewer: ```suggestion
`eslint-config-enact-proxy` needs to be installed locally on a project to enable in-editor linting:
```

Author: OK

```sh
npm install --save-dev eslint-config-enact-proxy
```

> 

Reviewer: Please guide to add `eslintConfig` property in package.json.

Author: OK

Ever since ESLint 6, global installs of ESLint configs are no longer supported.
To work around this new limitation, while still supporting in-editor linting, we've created a new [eslint-config-enact-proxy](https://github.com/enactjs/eslint-config-enact-proxy) package.
The [eslint-config-enact-proxy](https://github.com/enactjs/eslint-config-enact-proxy) acts like a small proxy config, redirecting ESLint to use a globally-installed Enact ESLint config.

```sh

> npm install eslint-config-enact-proxy

Reviewer: Why did you put this here?

Author: I will revert this change

Author: I think some developers may use this example as a usage guide of `eslint-config-enact-proxy`.
Installing `eslint-config-enact-proxy` is also an important step but there's no guidance for installing it.

```

In order for in-editor linting to work with our updated ESLint config, you'll need to upgrade to ESLint 7 or later. This can be installed globally by running:

```sh

> npm install --location=global eslint @enact/cli

Reviewer: This change seems not necessary.

Author: I will revert this change

```

Then, you will need to uninstall any previous globally-installed Enact linting package (everything but eslint itself):

```sh

> npm remove --location=global eslint-plugin-react eslint-plugin-react-hooks eslint-plugin-babel @babel/eslint-parser eslint-plugin-jest eslint-plugin-enact eslint-config-enact

Reviewer: Could you please add some references to changing `-g` option?
Also, `npm uninstall` seems proper here.

Author: OK

Install the [vscode-eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) plugin.  More information on linters and Visual Studio Code can be found [here](https://code.visualstudio.com/docs/languages/javascript#_linters).

> > **NOTE**: To open the Settings editor, use the following VS Code menu command:

Author: I think these comments are not needed because the upper link includes these comments.
